### PR TITLE
Support item configuration

### PR DESCRIPTION
**Pull request by aounhaider1**
1) Allow add/update/delete on Items.
2) Allow creation of an item while linking a channel.

fixes https://github.com/eclipse/smarthome/issues/1488, https://github.com/eclipse/smarthome/issues/1490

### CODE REVIEW

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/app.js [hunk 1]
```diff
@@ -52,6 +52,15 @@ angular.module('PaperUI', [ 'PaperUI.controllers', 'PaperUI.controllers.control'
         templateUrl : 'partials/configuration.html',
         controller : 'ConfigurationPageController',
         title : 'Configuration'
+    }).when('/configuration/items', {
+        templateUrl : 'partials/configuration.html',
+        controller : 'ConfigurationPageController',
+        title : 'Configuration'
+    }).when('/configuration/item/edit/:itemName', {
+        templateUrl : 'partials/item.config.html'
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/app.js]: Every other non-redirect URL contains a title, why not this one?

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            hasBackdrop : true,
+            locals : {
+                item : item
+            }
+        }).then(function() {
+            $scope.refresh();
+        });
+    }
+    $scope.getSrcURL = function(category, type) {
+        return category ? 'http://localhost:8080/icon/' + category.toLowerCase() : type ? 'http://localhost:8080/icon/' + type.toLowerCase().replace('item', '') : '';
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js]: Is it necessary to use the full URL here? I don't think it is a good idea to force the usage of 8080 as port to have a full functional Paper UI.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    itemService.getNonRecursiveAll(function(items) {
+        $scope.items = items;
+        if (itemName) {
+            var items = $filter('filter')(items, {
+                name : itemName
+            });
+            if (items.length > 0) {
+                $scope.item = items[0];
+                $scope.configMode = "edit";
+                $scope.srcURL = $scope.item.category ? 'http://localhost:8080/icon/' + $scope.item.category.toLowerCase() : $scope.item.type ? 'http://localhost:8080/icon/' + $scope.item.type.toLowerCase().replace('item', '') : ''
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js]: see above

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/shared.properties.js [hunk 4]
```diff
@@ -132,7 +133,47 @@ angular.module('PaperUI.controllers.rules').service('sharedProperties', function
...
+        value : "Color"
+    }, {
+        name : "ImageItem",
+        value : "Image"
+    }, {
+        name : "PlayerItem",
+        value : "Player"
+    }, {
+        name : "LocationItem",
+        value : "LocationItem"
```
Comment (maggu2810) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/shared.properties.js]: The LocationItem is the only one that name and value is equal. The other values consists of the name without the leading "Item". Has this done by intention?

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/app.js (hunk 1 above)
Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/app.js]: Actually title is unnecessary in some other routes as well as we are dynamically setting it anyway in the controller. e.g. thing/edit route sets title to 'Configuration' but it is removed as we set the path to thing UID inside its controller.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js (hunk 2 above)
Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.items.js]: You are right that is a mistake.

File: extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/shared.properties.js (hunk 4 above)
Comment (aounhaider1) [inline extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/shared.properties.js]: I will change that.

**Comment by aounhaider1** [comment]
Pull request updated accordingly.

**Comment by kaikreuzer** [comment]
Thanks, works nicely!

**Pull request merged**